src/EigenvalueSolverTypes.f90

```diff
@@ -37,6 +37,14 @@ MODULE EigenvalueSolverTypes
 #ifdef FUTILITY_HAVE_Trilinos
 USE ForTeuchos_ParameterList
 #endif
+
+#ifdef FUTILITY_HAVE_PETSC
+#include <petscversion.h>
+#if ((PETSC_VERSION_MAJOR>=3) && (PETSC_VERSION_MINOR>6))
```

Comment:
I'll add that I am just checking that `PETSC_VERSION_MINOR > 6`, but I should ideally check which version of PETSc actually switched to build modules and go from there.

Reply:
Yeah.  And maybe for good measure add an `|| PETSC_VERSION_MAJOR >= 4` just to attempt to future proof it a bit.

Reply:
reasonable